mypy/checker.py

```diff
@@ -1360,16 +1360,22 @@ def check_method_override_for_base_with_name(
             # it can be checked for compatibility.
             original_type = base_attr.type
             original_node = base_attr.node
+            if isinstance(original_node, TypeAlias):
+                # Overriding a type alias is always an error, it is static by nature.
+                self.msg.cannot_overide_alias(original_node.name(), base.name(), defn)
+                return False
             if original_type is None:
                 if self.pass_num < self.last_pass:
-                    # If there are passes left, defer this node until next pass,
-                    # otherwise try reconstructing the method type from available information.
+                    # If there are passes left, defer this node until next pass...
                     self.defer_node(defn, defn.info)
                     return True
+                # ...otherwise try reconstructing the supertype from available information.
                 elif isinstance(original_node, (FuncDef, OverloadedFuncDef)):
                     original_type = self.function_type(original_node)
                 elif isinstance(original_node, Decorator):
                     original_type = self.function_type(original_node.func)
+                elif isinstance(original_node, Var):
+                    original_type = AnyType(TypeOfAny.implementation_artifact)
```

Comment:
It's not clear to me how this is related to the title of the PR, but the description seems to indicate that this fixes another, related issue with `Var` nodes. Maybe tweak the title of the PR? 

Also, this looks like it could cause false negatives. It's better than crashing, but maybe we should generate a "Cannot determine type" error instead? At least there should be a comment about this, and potentially a follow-up issue.